tools/cldr-apps/js/src/esm/cldrVoteAgree.js

```diff
@@ -0,0 +1,107 @@
+ * including the assumption that the page and row for the path in question need to be visible when
+ * a vote is submitted. As the related modules get modernized it should become possible to avoid the
+ * heavy-handed use of cldrLoad.reloadV, and passing the DOM element "tr" to cldrVote.handleWiredClick.
+ *
+ * "Agree" buttons occur in forum posts in two GUI contexts: (1) the Info Panel shown on the
+ * right side of the Page table; (2) the Forum special page (with #forum in the URL). The row is
+ * normally already visible in the first context, but not in the second. The current implementation
+ * is intended to work in both contexts.
+ */
+import * as cldrLoad from "./cldrLoad.js";
+import * as cldrStatus from "./cldrStatus.js";
+import * as cldrSurvey from "./cldrSurvey.js";
+import * as cldrText from "./cldrText.js";
+import * as cldrVote from "./cldrVote.js";
+
+import { notification } from "ant-design-vue";
+
+/**
+ * A map from locale and path to values for which the user has agreed to vote
+ *
+ * A mapping gets added when the user presses an Agree button, then removed when
+ * the row in question is loaded and a vote is submitted
+ */
+const rememberToVote = {};
+
+/**
+ * The user has pressed a Forum "Agree" button in either the Info Panel or the
+ * special #forum page
+ *
+ * Arrange to load the page view for this path and then vote for the given value
+ *
+ * @param {String} locale - the locale in which to vote
+ * @param {String} xpstrid - the hex xpath id
+ * @param {String} value - the value to vote for
+ */
+function vote(locale, xpstrid, value) {
+  rememberToVote[agreeKey(locale, xpstrid)] = value;
+  cldrStatus.setCurrentLocale(locale);
+  cldrStatus.setCurrentId(xpstrid);
+  cldrStatus.setCurrentSpecial("");
+  cldrStatus.setCurrentPage("");
+  // cldrLoad.reloadV will cause the page table to be loaded,
+  // and then cldrTable will call voteIfAgreed for each row
+  cldrLoad.reloadV();
+}
+
+/**
+ * If we're waiting to make an "Agree" vote for this row, schedule it to happen
+ *
+ * This is called from cldrTable when the row for each path is loaded
+ *
+ * @param {Element} tr - a DOM element to pass to handleWiredClick
+ * @param {Object} theRow - an object describing the row
+ */
+function voteIfAgreed(tr, theRow) {
+  const key = agreeKey(cldrStatus.getCurrentLocale(), theRow.xpstrid);
+  const value = rememberToVote[key];
+  if (!value) {
+    return;
+  }
+  delete rememberToVote[key];
+  for (let item of Object.values(theRow.items)) {
+    if (value === getValueOrInherited(item.value, theRow.inheritedValue)) {
+      // set a timeout, to allow the table to finish loading before submitting the vote
+      setTimeout(reallyVote(tr, theRow, value, item.valueHash), 1);
+      return;
+    }
+  }
+  notifyNotFound(value);
+}
+
+function reallyVote(tr, theRow, value, valueHash) {
+  const pseudoButton = {
+    value: value,
+  };
+  cldrVote.handleWiredClick(tr, theRow, valueHash, undefined, pseudoButton);
```

Comment:
this seems fragile…  why not just use the voting API?

POST `/api/voting/{locale}/row/{hexXpathHash}`

payload is: 

```json
{
  "value": "The Vote Value",
  "voteLevelChanged": 4
}
```

`voteLevelChanged` vote downgrade, can omit or null to use default.

Reply:
and then mark the row for reload somehow

Reply:
the comments at top of cldrVoteAgree.js try to explain some of the rationale

a fundamental question is, for the new agree-vote feature, do we re-use the existing system or implement a separate parallel system?

the existing voting system keeps track of which rows have votes pending; how many votes are pending; distinctive display styles for rows depending on their vote-pending status; when an all-row response is received, avoidance of replacing newer single-row data with older all-row data for a particular row; how a row is displayed in response to a vote-submission failure; ...

a parallel system would risk screwing up the existing system -- is there a vote pending on this row? yes, but the existing system doesn't know about it, so the display doesn't match the reality

the existing system doesn't use api/voting -- that's not actually used at all, yet, is it? we keep postponing modernizing the page view since it's such a large task, involving lot of highly complex tightly coupled systems

when the page view is modernized we won't have dependencies on data in the DOM; in the meantime we do, so in this pr i've tried to adapt to the current situation with minimal complexity

Reply:
@srl295 thanks! i'll go ahead and merge this and follow up soon with another PR or two:
* add a dialog along the lines of the suggestion by @stenshamn "You indicated agreement with Vetter N, and your vote for value Z will be changed from X to Y. Please note that only the vote for Z will be changed. Other vote changes needed for consistency with this value will have to be made by you individually for those items."
* possibly refactor to reduce entanglement with legacy code